### PR TITLE
[TU-215] DataGrid: border & rounded corners

### PR DESCRIPTION
**Pull request by driesd**
### Description

This PR adds a `bordered` boolean prop. If `true`, the `Datagrid` will have rounded corners and a border around it.

**Note:** I had to `revert` the `4 latest commits` that were meant to for a next PR. Sorry for that 🙈 

#### Screenshot before this PR

schermafdruk 2018-11-16 15 11 54

#### Screenshot after this PR

schermafdruk 2018-11-16 15 11 32

### Breaking changes

None.

### CODE REVIEW

File: src/components/datagrid/DataGrid.js [hunk 1]
```diff
@@ -13,35 +13,21 @@ import HeaderRow from './HeaderRow';
 import BodyRow from './BodyRow';
 import cx from 'classnames';
 import omit from 'lodash.omit';
-import throttle from 'lodash.throttle';
+import ReactResizeDetector from 'react-resize-detector';
```
Comment (lowiebenoot) [inline src/components/datagrid/DataGrid.js]: Does the resizing and overflowing stuff belong to this PR?

Comment (driesd) [reply src/components/datagrid/DataGrid.js]: Yeah, you're right @lowiebenoot . I'll close this PR and split it up in two separate (and less messy) PRs.

File: src/components/datagrid/DataGrid.js [hunk 2]
```diff
@@ -83,36 +69,44 @@ class DataGrid extends PureComponent {
...
   setCalculatedRowWidth = () => {
-    if (isElementOverflowingX(this.scrollableNode) && this.rowNodes) {
-      const rowDOMNodes = [];
-      let maxRowWidth = 0;
+    const rowDOMNodes = [];
+    let maxRowWidth = 0;
 
-      [...this.rowNodes.values()].filter(rowNode => rowNode != null).forEach(rowNode => {
-        const rowDOMNode = ReactDOM.findDOMNode(rowNode);
+    [...this.rowNodes.values()].filter(rowNode => rowNode != null).forEach(rowNode => {
```
Comment (kevinwaelkens) [inline src/components/datagrid/DataGrid.js]: Q: Doesn't `this.rowNodes.values().filter` suffice?
PP: `rowNode !== null`

File: src/components/datagrid/DataGrid.js [hunk 3]
```diff
@@ -83,36 +69,44 @@ class DataGrid extends PureComponent {
...
+      const rowDOMNode = ReactDOM.findDOMNode(rowNode);
 
-        if (rowDOMNode) {
-          const totalRowChildrenWidth = [...rowDOMNode.children]
-            .map(child => child.offsetWidth)
-            .reduce((accumulatedChildWidth, currentChildWidth) => accumulatedChildWidth + currentChildWidth);
+      if (rowDOMNode) {
+        const totalRowChildrenWidth = [...rowDOMNode.children]
```
Comment (kevinwaelkens) [inline src/components/datagrid/DataGrid.js]: Q: Here again then: `[...xxxx]` seems a redundant action to perform?

**Pull request closed without merging**